app/src/main/java/com/example/lapnguyen/weather_3/screen/mainscreen/MainActivity.java

```diff
@@ -0,0 +1,92 @@
+import com.example.lapnguyen.weather_3.api.WeatherService;
+import com.example.lapnguyen.weather_3.data.model.Weather;
+import com.example.lapnguyen.weather_3.data.source.MainRepository;
+import com.example.lapnguyen.weather_3.data.source.MainRepositoryImplement;
+import com.example.lapnguyen.weather_3.screen.settingscreen.SettingsActivity;
+import com.example.lapnguyen.weather_3.utils.Navigator;
+import java.util.Date;
+
+/**
+ * Created by lapnguyen on 14/06/2017.
+ */
+public class MainActivity extends AppCompatActivity implements MainContract.View {
+
+    private MainContract.Presenter mPresenter;
+    private TextView mCurrentLocation, mCurrentTemperature, mWindSpeed, mHumidityPercent, mUvIndex,
+            mLastUpdatedTime;
+    private ImageView mWeatherIcon;
+
+    Navigator mNavigator;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        initWidget();
+        mNavigator = new Navigator(this);
+        MainRepository mainRepository =
+                new MainRepositoryImplement(ServiceGenerator.createService(WeatherService.class));
+        mPresenter = new MainPresenter(this, mainRepository);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle item selection
+        switch (item.getItemId()) {
+            case R.id.settings:
+                mNavigator.startActivity(SettingsActivity.class);
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    public void initWidget() {
+        mCurrentLocation = (TextView) findViewById(R.id.text_currentLocation);
+        mCurrentTemperature = (TextView) findViewById(R.id.text_currentTemperature);
+        mWindSpeed = (TextView) findViewById(R.id.text_windSpeed);
+        mHumidityPercent = (TextView) findViewById(R.id.text_humidity_percent);
+        mUvIndex = (TextView) findViewById(R.id.text_uvIndex);
+        mWeatherIcon = (ImageView) findViewById(R.id.image_weather_icon);
+        mLastUpdatedTime = (TextView) findViewById(R.id.text_lastUpdate);
+    }
+
+    public String EpochConvert(long epoch) {
+        return DateFormat.format("dd/MM/yyyy HH:mm:ss", new Date(epoch * 1000)).toString();
+    }
+
+    @Override
+    public void LoadWeatherDetailSuccess(Weather weather) {
+        displayData(weather);
+    }
+
+    public void displayData(Weather weather) {
+
+        mLastUpdatedTime.setText(
+                "Last updated : " + EpochConvert(weather.getCurrently().getTime()));
+        mUvIndex.setText("UV Index : " + weather.getCurrently().getUvIndex());
+        mCurrentTemperature.setText("" + weather.getCurrently().getTemperature());
```

Comment:
WARNING: [SetTextI18n] Do not concatenate text displayed with `setText`. Use resource string with placeholders.
 When calling `TextView#setText`
* Never call `Number#toString()` to format numbers; it will not handle fraction separators and locale-specific digits properly. Consider using `String#format` with proper format specifications (`%d` or `%f`) instead.
* Do not pass a string literal (e.g. "Hello") to display text. Hardcoded text can not be properly translated to other languages. Consider using Android resource strings instead.
* Do not build messages by concatenating text chunks. Such messages can not be properly translated.